src/Sarif.Viewer.VisualStudio/ErrorList/SarifResultTableEntry.cs

```diff
@@ -22,14 +22,35 @@ namespace Microsoft.Sarif.Viewer.ErrorList
 {
     internal sealed class SarifResultTableEntry : ITableEntry, IDisposable
     {
+        internal const string SuppressionStateColumnName = "suppression";
+
         private bool isDisposed;
 
         private readonly Dictionary<string, object> columnKeyToContent = new Dictionary<string, object>(StringComparer.InvariantCulture);
 
-        public static readonly ReadOnlyCollection<string> SupportedColumns = new ReadOnlyCollection<string>(new[] {
+        // The set of columns we display depends on the contents of the displayed log file.
+        // Per https://github.com/microsoft/sarif-visualstudio-extension/issues/114, we do not
+        // display "suppressed" or "absent" issues by default.
+        //
+        // If we open a log file that contains a suppressed issue, we display the "Suppression
+        // State" column, setting its filter to NOT display the suppressed items. The appearance
+        // of the column is a hint to the user that there are suppressed results available to examine.
+        //
+        // Similarly, if we open a log file that contains absent issues, we display the "Category"
+        // column, setting its filter to NOT display the absent items. The appearance of the column
+        // is a hint to the user that there are absent results available to examine.
+        //
+        // There is not a direct mechanism to add or remove columns based on an individual entry.
+        // You can, however, add a new ITableDataSource (ITableManager.AddSource(source[, columns])
+        // where you can specify the columns you want to appear in the table (at which point VS adds
+        // the column to the table, though it won't be shown to the user unless it is visible).
+        // Removing the data source removes the column (although we don't have a requirement to do
+        // that), unless some other source also added the same column.
+
+        // The "basic" columns are always displayed.
+        public static readonly ReadOnlyCollection<string> BasicColumns = new ReadOnlyCollection<string>(new[] {
             StandardTableKeyNames2.TextInlines,
```

Comment:
These really should be the corresponding column names (StandardTableColumnDefinitions....), It doesn't matter for these -- since the desired column name matches the key name but it is confusing.

Reply:
Yes, you're right, it confused me. I'll change them.

---
In reply to: [538000615](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538000615) [](ancestors = 538000615)